models/iosxr/ntp_global/iosxr_ntp_global.yaml

```diff
@@ -0,0 +1,225 @@
+---
+module: iosxr_ntp_global
+short_description: Manages ntp resource module
+description: This module configures and manages the attributes of  ntp on Cisco
+  IOSXR platforms.
+version_added: 2.5.0
+author: Ashwini Mhatre (@amhatre)
+notes:
+  - Tested against IOSXR 7.0.2.
+  - This module works with connection C(network_cli).
+options:
+  config:
+    description: A dictionary of ntp options
+    type: dict
+    suboptions:
+        access_group:
+          description: Control NTP access
+          type: list
+          elements: dict
+          suboptions:
+            ipv4: &ipv4
+              type: dict
+              description: Configure IPv4 access
+              suboptions:
+                peer: &peer
```

Comment:
`peer`, `query_only, `serve`, `serve_only` are lists in NX-OS. Is it not the same for XR?

Reply:
`drift aging time 0
 access-group vrf t1 ipv4 peer test
 access-group vrf t1 ipv4 serve test
 access-group vrf t1 ipv4 query-only test1
 access-group vrf test1 ipv4 serve test
 access-group ipv4 peer test2
 access-group ipv4 serve test
 access-group ipv4 serve-only test
 access-group ipv6 peer test1
 source GigabitEthernet0/0/0/0
 broadcastdelay 1
 log-internal-sync
!

RP/0/RP0/CPU0:ios#configure              
Mon Aug 30 16:05:18.415 UTC
RP/0/RP0/CPU0:ios(config)#access-group ipv4 peer test3
                             ^
% Invalid input detected at '^' marker.
RP/0/RP0/CPU0:ios(config)#ntp access-group ipv4 peer test3
RP/0/RP0/CPU0:ios(config)#ntp access-group ipv4 serve test1
RP/0/RP0/CPU0:ios(config)#ntp access-group ipv4 query-only test2
RP/0/RP0/CPU0:ios(config)#commit
Mon Aug 30 16:06:06.389 UTC
RP/0/RP0/CPU0:ios(config)#exit
RP/0/RP0/CPU0:ios#show running-config ntp
Mon Aug 30 16:06:12.246 UTC
ntp
 max-associations 10
 interface GigabitEthernet0/0/0/0
  multicast client 224.0.0.9
  multicast key 1
  multicast destination 224.0.0.8
 !
 authentication-key 1 md5 encrypted test
 authenticate
 trusted-key 1
 ipv4 dscp af11
 ipv6 precedence routine
 peer vrf test1 1.1.1.3 iburst
 server vrf test1 1.1.1.4 burst
 server 1.1.1.1 iburst
 peer 1.1.1.2
 peer 1.1.1.3 version 2
 drift file apphost
 drift aging time 0
 access-group vrf t1 ipv4 peer test
 access-group vrf t1 ipv4 serve test
 access-group vrf t1 ipv4 query-only test1
 access-group vrf test1 ipv4 serve test
 access-group ipv4 peer test3
 access-group ipv4 serve test1
 access-group ipv4 serve-only test
 access-group ipv4 query-only test2
 access-group ipv6 peer test1
 source GigabitEthernet0/0/0/0
 broadcastdelay 1
 log-internal-sync
!

RP/0/RP0/CPU0:ios#
`

Reply:
no this is not the case in iosxr. we can only configure one peer, query_only, serve, serve_only  at time

Reply:
Ack.